code/modules/mapping/space_management/space_reservation.dm

```diff
@@ -56,6 +70,42 @@
 		SSmapping.unused_turfs["[cordon_turf.z]"] -= cordon_turf
 		SSmapping.used_turfs[cordon_turf] = src
 
+	//swap the area with the pre-cordoning area
+	for(var/turf/pre_cordon_turf as anything in pre_cordon_turfs)
+		make_repel(pre_cordon_turf)
+
+///Register signals in the cordon "danger zone" to do something with whoever trespasses
+/datum/turf_reservation/proc/make_repel(turf/pre_cordon_turf)
+	SHOULD_CALL_PARENT(TRUE)
+	//Okay so hear me out. If we place a special turf IN the reserved area, it will be overwritten, so we can't do that
+	//But signals are preserved even between turf changes, so even if we register a signal now it will stay even if that turf is overriden by the template
+	RegisterSignals(pre_cordon_turf, list(COMSIG_PARENT_QDELETING, COMSIG_TURF_RESERVATION_RELEASED), PROC_REF(on_stop_repel))
+
+/datum/turf_reservation/proc/on_stop_repel(turf/pre_cordon_turf)
+	SHOULD_CALL_PARENT(TRUE)
+	SIGNAL_HANDLER
+
+	stop_repel(pre_cordon_turf)
+
+///Unregister all the signals we added in RegisterRepelSignals
+/datum/turf_reservation/proc/stop_repel(turf/pre_cordon_turf)
+	UnregisterSignal(pre_cordon_turf, list(COMSIG_PARENT_QDELETING, COMSIG_TURF_RESERVATION_RELEASED))
+
+/datum/turf_reservation/transit/make_repel(turf/pre_cordon_turf)
+	..()
+
+	RegisterSignal(pre_cordon_turf, COMSIG_ATOM_ENTERED, PROC_REF(space_dump))
+
+/datum/turf_reservation/stop_repel(turf/pre_cordon_turf)
```

Comment:
Why are we declaring the `stop_repel` proc twice for `/datum/turf_reservation`? Missing the subtype?

Reply:
Missing the subtype.